why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,202 @@
+:::
+
+When working with data analysis, data is rarely ready to be used 'as
+is'. There will always be things that are either very wrong (like text
+in what should be a number field), slightly wrong (like a single missing
+or incorrect value), or an outlier potentially due to an error in data
+entry. Since we don't have domain knowledge to know about missing values
+or the definition or meaning of an outlier. But there are some basic
+things we can check and fix, such as whether the data contradicts any
+constraints in the definition of the data resource (eg. a number in a
+date field). There are a lot of tools that can be used to both analyse a
+data set, and to clean it, either as part of the analysis, or as part of
+the initial check before the data is transformed and stored. So, our
+question is:
+
+*Which tool should we use for cleaning and processing data in our
+software and what tool should we recommend users to use when cleaning
+and processing needs to happen outside of our software?*
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+-   Can be installed and used in Python.
+-   Is designed and built to be fast for processing data.
+-   Has minimal dependencies, so can be installed quickly.
+-   Is flexible to reading different data storage formats (e.g. SQL
+    databases, Parquet, CSV).
+-   Can write to different storage formats (as with above).
+-   We need a tool that can clean data and is either written in Python
+    or has a Python package available.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+While there are a number of tools available to use for data cleaning,
+given our requirements, these are the options we've considered:
+
+-   [DuckDB](https://www.duckdb.org/)
+-   [Pandas](https://pandas.pydata.org/)
+-   [Polars](https://pola.rs/)
+
+### DuckDB
+
+DuckDB is an in-process online analytical process (OLAP) database tool
+that has API extensions in many languages including Python. It is a very
+young tool, version 1.0.0 was released in 2024, although the first
+version was released in 2019. The developers have set it up as a
+not-for-profit foundation in the Netherlands.
+
+::: columns
+::: column
+#### Benefits
+
+-   Although a very young project it is already in use by a number of
+    large companies and has a large community of users.
+    -   The code is open source, and the way it has been set up is
+        promising for it to remain that way.
+-   It is very fast, and can handle large data sets.
+-   It is designed for data cleaning using SQL syntax.
+-   Extensive documentation with user guide sections embedded.
+:::
+
+::: column
+#### Drawbacks
+
+-   It is not designed exclusively to be used as a data cleaning tool,
+    this is more of an add-on on the way to storing data.
```

Comment:
```suggestion
```

Given that DuckDB doesn't store data in any specific format and instead imports from other formats (see https://duckdb.org/docs/data/data_sources), this isn't true. Combined with changing the terminology to processing rather than cleaning, I hope this makes more sense and that we can remove this drawback.